specs/lib/objSpec.js

```
@@ -0,0 +1,328 @@
+var objObjectsUrl = 'specs/data/box-objects/box-objects.obj';
+var objGroupsUrl = 'specs/data/box-groups/box-groups.obj';
+var objObjectsGroupsUrl = 'specs/data/box-objects-groups/box-objects-groups.obj';
+var objUsemtlUrl = 'specs/data/box-usemtl/box-usemtl.obj';
+var objNoMaterialsUrl = 'specs/data/box-no-materials/box-no-materials.obj';
+var objMultipleMaterialsUrl = 'specs/data/box-multiple-materials/box-multiple-materials.obj';
+var objUncleanedUrl = 'specs/data/box-uncleaned/box-uncleaned.obj';
+var objMtllibUrl = 'specs/data/box-mtllib/box-mtllib.obj';
+var objMissingMtllibUrl = 'specs/data/box-missing-mtllib/box-missing-mtllib.obj';
+var objTexturedUrl = 'specs/data/box-textured/box-textured.obj';
+var objMissingTextureUrl = 'specs/data/box-missing-texture/box-missing-texture.obj';
+var objSubdirectoriesUrl = 'specs/data/box-subdirectories/box-textured.obj';
+var objComplexMaterialUrl = 'specs/data/box-complex-material/box-complex-material.obj';
+var objInvalidContentsUrl = 'specs/data/box/box.mtl';
+var objInvalidUrl = 'invalid.obj';
+
+function getMeshes(data) {
+    var meshes = [];
+    var nodes = data.nodes;
+    var nodesLength = nodes.length;
+    for (var i = 0; i < nodesLength; ++i) {
+        meshes = meshes.concat(nodes[i].meshes);
+    }
+    return meshes;
+}
+
+function getPrimitives(data) {
+    var primitives = [];
+    var nodes = data.nodes;
+    var nodesLength = nodes.length;
+    for (var i = 0; i < nodesLength; ++i) {
+        var meshes = nodes[i].meshes;
+        var meshesLength = meshes.length;
+        for (var j = 0; j < meshesLength; ++j) {
+            primitives = primitives.concat(meshes[j].primitives);
+        }
+    }
+    return primitives;
+}
+
+function getImagePath(objPath, relativePath) {
+    return path.normalize(path.join(path.dirname(objPath), relativePath));
+}
+
+describe('obj', function() {
+    it('loads obj with positions, normals, and uvs', function(done) {
+        expect(loadObj(objUrl)
```

In that way if the promise failed it wouldn't reach `done()`. The test would still fail due to a timeout, but the version in `objSpec` would fail faster.